cmd/tpm-manager/tpm-manager.go

```
@@ -0,0 +1,507 @@
+		reverify, err := strconv.Atoi(configmap.Data["reverify"])
+		if err == nil {
+			manager.recurring = reverify
+			// Trigger the reverification logic. If it's already in the
+			// middle of reverifying, drop the event - it'll handle it
+			// at the end of reverification.
+			select {
+			case manager.recurringChan <- reverify:
+			default:
+			}
+		}
+	}
+}
+
+func run(stop <-chan struct{}) {
+	_, nodeController := framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return manager.client.Nodes().List(options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return manager.client.Nodes().Watch(options)
+			},
+		},
+		&api.Node{},
+		controller.NoResyncPeriodFunc(),
+		framework.ResourceEventHandlerFuncs{
+			AddFunc: nodeAddFn,
+		},
+	)
+
+	_, configController := framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return manager.client.ConfigMaps("kube-system").List(options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return manager.client.ConfigMaps("kube-system").Watch(options)
+			},
+		},
+		&api.ConfigMap{},
+		controller.NoResyncPeriodFunc(),
+		framework.ResourceEventHandlerFuncs{
+			AddFunc:    configAddFn,
+			UpdateFunc: configUpdateFn,
+		},
+	)
+
+	_, policyController := framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return manager.tpmhandler.PolicyClient.List(&options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return manager.tpmhandler.PolicyClient.Watch(&options)
+			},
+		},
+		&runtime.Unstructured{},
+		controller.NoResyncPeriodFunc(),
+		framework.ResourceEventHandlerFuncs{
+			AddFunc:    policyAddFn,
+			UpdateFunc: policyUpdateFn,
+			DeleteFunc: policyDeleteFn,
+		},
+	)
+	loadPolicy()
```

ignoring errors from this function